### ISSUE TITLE
Bug: Handle comments in block-spacing

### ISSUE THREAD

**Issue by alberto** [eslint/eslint#4387]
`block-spacing` should handle comments the same way `space-in-parens` does, as discussed in #4302.

Here is an example of a valid case:
```js
{
  code: "if (a) {/* comment */ foo(); /* comment */}", 
  options: ["never"]
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.
